packages/engine/src/experiment/controller/run.rs

```diff
@@ -346,3 +349,36 @@ fn worker_pool_exit_logic(
 
     false
 }
+
+/// Shared memory cleanup in the process hard crash case.
+/// Not required for pod instances.
+pub fn cleanup_experiment(experiment_id: &ExperimentId) {
+    if let Err(err) = MemoryId::clean_up(experiment_id) {
+        tracing::warn!("{}", err);
+    }
+
+    if let Err(err) = cleanup_python_runner(experiment_id) {
+        tracing::warn!("{}", err);
+    }
+
+    remove_experiment_parts(experiment_id);
+}
+
+// We might want to move the "/parts" folder to a temporary folder (like "/tmp" or "/var/tmp").
+fn remove_experiment_parts(experiment_id: &ExperimentId) {
+    let path = format!("{RELATIVE_PARTS_FOLDER}/{experiment_id}");
+    match std::fs::remove_dir_all(&path) {
+        Ok(_) => {
+            tracing::trace!(
+                experiment = %experiment_id,
+                "Removed parts folder for experiment {experiment_id}: {path:?}"
+            );
+        }
+        Err(err) => {
+            tracing::warn!(
+                experiment = %experiment_id,
+                "Could not clean up {path:?}: {err}"
+            );
+        }
+    }
+}
```

Comment:
`RELATIVE_PARTS_FOLDER` was only known to `output::buffer` before. I'm not very happy about the `output::buffer` module at all, but removing the parts should happen in the same module where they are created.
The comment may be better suited on `RELATIVE_PARTS_FOLDER`.

Reply:
Moved the comment in https://github.com/hashintel/hash/pull/494/commits/b9390bce74bfb6383e5722e26ef6ee5adf9c1fe6